src/Records/Types/DomainName.php

```diff
@@ -87,12 +89,8 @@ public function setLabels(array $labels, $tldFirst = false)
         $length = $count = 0;
 
         foreach ($labels as &$label) {
-            $label = \LibDNS\normalize_name($label);
-            $labelLength = \strlen($label);
-            if ($labelLength > 63) {
-                throw new \InvalidArgumentException('Label list is not a valid domain name: Label ' . $label . ' length exceeds 63 byte limit');
-            }
-            $length += $labelLength + 1;
+            $label = (string)new NormalizedLabel($label);
```

Comment:
But where's the gain? It's not used as type anywhere, it isn't passed around, it's just normalized and used.